### DISCUSSION TITLE
Signing/Validation of custom Python code

### DISCUSSION THREAD

**Discussion by JHancox** [NVIDIA/NVFlare#915] [Ideas]
One of our customers commented that the ability to execute custom python code at various points in the process presents a major security threat (e.g. for Healthcare customers). Obviously there are ways to lock down what is and is not possible in NVFlare, but one alleviation might be to allow uploaded python modules to be signed by the submitting site and verified prior to execution on the server or client, so that any change to the script will invalidate that client or server (in the case of an aggregation script).

**Comment by YuanTingHsieh** [comment]
Thanks for the feedback/idea!

We will be discussing this and let you know whether we will add it in next or future releases.

**Comment by chesterxgchen** [comment]
We will include this in next release
